src/hooks/useApproveAndSwap.ts

```diff
@@ -65,18 +67,26 @@ export function useApproveAndSwap(): (
     transactionDeadlineSelected,
     infiniteApproval,
   } = useSelector((state: AppState) => state.user)
+
   return async function approveAndSwap(
     state: ApproveAndSwapStateArgument,
   ): Promise<void> {
     try {
-      if (!account) throw new Error("Wallet must be connected")
+      if (!account || !library) throw new Error("Wallet must be connected")
       if (state.swapType === SWAP_TYPES.DIRECT && !state.swapContract)
         throw new Error("Swap contract is not loaded")
       if (state.swapType !== SWAP_TYPES.DIRECT && !state.bridgeContract)
         throw new Error("Bridge contract is not loaded")
       if (chainId === undefined) throw new Error("Unknown chain")
       // For each token being deposited, check the allowance and approve it if necessary
-      const tokenContract = tokenContracts?.[state.from.symbol] as Erc20
+      const token = tokensMap[state.from.symbol]
+      if (!token) throw new Error("Token not found")
+      const tokenContract = getContract(
```

Comment:
Awesome! This is so much better. Nit, no need to lowercase address here. The only reason we do that elsewhere is to make addresses uniform for equality checks